test/unit/custom_header_test.rb

```diff
@@ -23,6 +23,24 @@ def test_can_set_custom_headers
     end
   end
 
+  def test_can_set_custom_accept_headers
+    stub_request(:get, "http://example.com/custom_header_resources/1")
+      .with(headers: {"Accept" => "application/vnd.api+json,application/vnd.api+json;version=2"})
+      .to_return(headers: {accept: "application/vnd.api+json,application/vnd.api+json;version=2", content_type: "application/vnd.api+json"}, body: {
+        data: {
+          type: "custom_header_resources",
+          id: "1",
+          attributes: {
+            title: "My Blooming Test!"
+          }
+        }
+      }.to_json)
+
+    CustomHeaderResource.with_headers(accept: "application/vnd.api+json,application/vnd.api+json;version=2") do
```

Comment:
should this test omit setting the application/vnd.api+json to ensure that the middleware is updating it?

Reply:
That should probably be an additional test, as my code allows for the inclusion/exclusion of application/vnd.api+json